src/config.js

```diff
@@ -0,0 +1,33 @@
+const mergeOptions = require('merge-options')
+const minimatch = require('minimatch')
+
+const getConfigForFunction = ({ config, func }) => {
+  if (!config) {
+    return {}
+  }
+
+  // It's safe to mutate the array because it's local to this function.
+  // eslint-disable-next-line fp/no-mutating-methods
+  const matches = Object.keys(config)
+    .filter((expression) => minimatch(func.name, expression))
+    .map((expression) => {
+      const wildcardCount = [...expression].filter((char) => char === '*').length
+
+      // The weight increases with the number of hardcoded (i.e. non-wildcard)
+      // characters — e.g. "netlify" has a higher weight than "net*". We do a
+      // subtraction of 1 if there is at least one wildcard character, so that
+      // "netlify" has a higher weight than "netlify*".
+      const weight = expression.length - wildcardCount - (wildcardCount ? 1 : 0)
+
+      return {
+        expression,
+        weight,
+      }
+    })
+    .sort((fA, fB) => fA.weight - fB.weight)
```

Comment:
When two functions have the same `weight`, should we sort them according to a second criteria?

Reply:
When they have the same weight, we return `0` in the sorting function, which leaves the elements unchanged. Do you feel that is not sufficient?

Reply:
Is the original sorting order based on the return value from `fs.readdir()`?

Since the sorting order is significant due to the merging, I am wondering whether we should have a more explicit/defined second sorting order?

Reply:
> Is the original sorting order based on the return value from fs.readdir()?

The original sorting order is defined by the order of the keys of the configuration object.

(We're comparing expression objects, not functions,  which made me realise that the `fA` and `fB` variable names are misleading. I'll update that.)

As for a second sorting order, what do you have in mind?

Reply:
The order of the keys of the configuration object sounds like a good "second" sorting order, what do you think? This is probably what the user would expect.

If this is the current behavior, then we're good :+1:

Reply:
Agreed! 👍🏻